languagetool-language-modules/de/src/test/java/org/languagetool/rules/de/AgreementRuleTest.java

```diff
@@ -54,7 +54,7 @@ public void testCompoundMatch() throws IOException {
     assertBad("Doch dieser sehr kleine Magnesium Anteil ist entscheidend.", "dieser sehr kleine Magnesiumanteil", "dieser sehr kleine Magnesium-Anteil");
     assertBad("Die Standard Priorität ist 5.", "Die Standardpriorität", "Die Standard-Priorität");
     assertBad("Die derzeitige Standard Priorität ist 5.", "Die derzeitige Standardpriorität", "Die derzeitige Standard-Priorität");
-    assertBad("Ein neuer LanguageTool Account", "Ein neuer LanguageTool-Account");
+    assertBad("Ein neuer LanguageTool Account", "Ein neuer LanguageToolaccount", "Ein neuer LanguageTool-Account");
```

Comment:
This doesn't look right at all, but that follows from the code's logic. `HunspellRule#addIgnoreWords` adds this `LanguageTool` word. Previously `updateIgnoredWordDictionary` wasn't called after that, so it wasn't considered in `startsWithIgnoredWord`, which looks like a bug to me. Now `startsWithIgnoredWord` always uses all ignored words, and the behavior is changed. It probably should be addressed in some way. If you believe that previous behavior was correct and Hunspell's ignored words shouldn't be checked as prefixes, I can adapt the patch to that.

Reply:
Indeed it doesn't look right, but I don't have time to look into this now.

Reply:
OK, I'll work around that

Reply:
Done. Now there are no regressions in tests.